Restaurant/App.xaml.cs

```diff
@@ -16,63 +18,74 @@
-    public partial class App : Application
-    {
-        private const string CONNECTION_STRING = "Data Source=restaurant.db";
-        private readonly Restaurant _restaurant;
-        private readonly RestaurantDbContextFactory _restaurantDbContextFactory;
-        private readonly NavigationStore _navigationStore;
-        private readonly MessageStore _messageStore;
-        private readonly MessageViewModel _messageViewModel;
-        private readonly ILoggerFactory _loggerFactory;
+	/// <summary>
+	/// Interaction logic for App.xaml
+	/// </summary>
+	public partial class App : Application
+	{
+		private const string CONNECTION_STRING = "Data Source=restaurant.db";
+		private readonly IHost _host;
 
-        public App()
-        {
-            _restaurantDbContextFactory = new RestaurantDbContextFactory(CONNECTION_STRING);
-            var databaseDishProvider = new DatabaseDishProvider(_restaurantDbContextFactory);
-            var databaseOrderCreator = new DatabaseOrderCreator(_restaurantDbContextFactory);
-            var databaseOrdersProvider = new DatabaseOrdersProvider(_restaurantDbContextFactory);
+		public App()
+		{
+			_host = Host.CreateDefaultBuilder()
+					.UseSerilog((host, loggerConfiguration) =>
+					{
+						loggerConfiguration.WriteTo.File(@"..\Logs\log.txt", rollingInterval: RollingInterval.Day);
+					})
+					.ConfigureServices(services =>
+					{
+						services.AddSingleton(new RestaurantDbContextFactory(CONNECTION_STRING));
+						services.AddSingleton<DatabaseDishProvider>();
+						services.AddSingleton<DatabaseOrderCreator>();
+						services.AddSingleton<DatabaseOrdersProvider>();
 
-            _restaurant = new Restaurant("Panorama", databaseDishProvider, databaseOrderCreator, databaseOrdersProvider);
-            _navigationStore = new NavigationStore();
-            _messageStore = new MessageStore();
-            _messageViewModel = new MessageViewModel(_messageStore);
-            _loggerFactory = LoggerFactory.Create(builder =>
-            {
-                var loggingConfiguration = new LoggerConfiguration()
-                .WriteTo.File(@"..\Logs\log.txt", rollingInterval: RollingInterval.Day);
-                builder.AddSerilog(loggingConfiguration.CreateLogger());
-            });
-        }
+						services.AddSingleton((s) => new Restaurant(
+							"Panorama",
+							s.GetRequiredService<DatabaseDishProvider>(),
+							s.GetRequiredService<DatabaseOrderCreator>(),
+							s.GetRequiredService<DatabaseOrdersProvider>()));
 
-        protected override async void OnStartup(StartupEventArgs e)
-        {
-            using (var dbContext = _restaurantDbContextFactory.CreateDbContext())
-            {
-                dbContext.Database.Migrate();
-            }
-            MainWindow = new MainWindow()
-            {
-                DataContext = new MainViewModel(_navigationStore)
-            };
-            MainWindow.Show();
+						services.AddSingleton<MessageViewModel>();
```

Comment:
Why are all services Singleton? 
Singleton always lives in memory and never frees the memory. When we open  Main Chef View,  we don't need Database Dish Provider in memory, so we can remove it (The Garbage Collector does it). And many others services have to be Transient.
Always start with Transient services. To make Singleton service, you must have the real big reasons.
I see Singleton only for MainWindow and MainViewModel, and for Func<MenuAndBasketViewModel> and Func<MainChefViewModel>, because
1.  MainWindow and MainViewModel - always exist, and only one instance. 
2. Func<....> just creates a new instance of ...ViewModel.